libeos-updater-util/flatpak.c

```diff
@@ -1457,6 +1457,651 @@ euu_filter_for_existing_flatpak_ref_actions (GHashTable *ref_actions  /* (elemen
                                            progresses);
 }
 
+static guint
+flatpak_remote_hash (gconstpointer data)
+{
+  FlatpakRemote *remote = FLATPAK_REMOTE (data);
+  /* The casting dance here is because want to g_direct_hash the priority
+   * which may be a different size to gconstpointer. We don't really care
+   * too much about potential overflow - we just want it to hash
+   * consistently. */
+  gsize prio = (gsize) flatpak_remote_get_prio (remote);
+  const gchar *name = flatpak_remote_get_name (remote);
+  const gchar *collection_id = flatpak_remote_get_collection_id (remote);
+
+  return (g_direct_hash ((gconstpointer) prio) ^
+          g_str_hash (name) ^
+          ((collection_id != NULL) ? g_str_hash (collection_id) : 0));
+}
+
+static gboolean
+flatpak_remote_equal (gconstpointer a,
+                      gconstpointer b)
+{
+  FlatpakRemote *a_remote = FLATPAK_REMOTE (a), *b_remote = FLATPAK_REMOTE (b);
+  return (flatpak_remote_get_prio (a_remote) == flatpak_remote_get_prio (b_remote) &&
+          g_strcmp0 (flatpak_remote_get_name (a_remote),
+                     flatpak_remote_get_name (b_remote)) == 0 &&
+          g_strcmp0 (flatpak_remote_get_collection_id (a_remote),
+                     flatpak_remote_get_collection_id (b_remote)) == 0);
+}
+
+static gboolean
+list_all_remote_refs_in_flatpak_installation (FlatpakInstallation  *installation,
+                                              GPtrArray           **out_remotes_priority_order,
+                                              GHashTable          **out_refs_for_remotes,
+                                              GCancellable         *cancellable,
+                                              GError              **error)
+{
+  g_autoptr(GPtrArray) remotes = NULL;
+  g_autoptr(GHashTable) refs_for_remotes = NULL;
```

Comment:
Note: I think I’m going to relax the requirement for declarations to be at the top of the function in eos-updater code. Having written a few projects with inline declarations now, I can say I’m a convert. I don’t expect you to change any of this code to use inline declarations though — just an FYI for future patches!

Reply:
(For future reference, this was done as #192.)